### ISSUE TITLE
Couldn't open wallet: invalid signature

### ISSUE THREAD

**Issue by elibroftw** [monero-project/monero-gui#3874]
![image](https://user-images.githubusercontent.com/21298211/160743866-54e377ff-249c-4254-9861-c952c7051aec.png)

Try to open a wallet and encountering this. Not sure if it's because I had to restart my computer via PC button today.

**Comment by elibroftw** [comment]
Already replaced the wallet files with a backup so can't do more testing.

**Comment by selsta** [comment]
It means the wallet cache is corrupted, seems to be related to your force shutdown.

**Comment by elibroftw** [comment]
Okay then can that error message be updated?

**Comment by elibroftw** [comment]
Or better yet, after every wallet save or auto save, a backup file should be generated. Then in these cases where the file is corrupted, just look for a backup file and overwrite the wallet file with it.

**Comment by elibroftw** [comment]
And I guess document what auto save actually does and why a user would want it enabled or disabled. What does the wallet do or not do if auto save is disabled?

**Comment by selsta** [comment]
Autosave means it automatically saves the wallet every x minutes. Otherwise it only saves on exit and after a transaction if I remember correctly.

**Comment by selsta** [comment]
Yes, scanned blocks get saved in the wallet cache. If you close without saving it has to rescan from the last saved height.

**Comment by MaxMusienko** [comment]
Hi all! I have a similar situation today. Please tell me how to solve the problem? Thank you!

**Comment by elibroftw** [comment]
You can either restore your wallet from seed, or use a backup file. Since I have two devices, I just replaced my local wallet files with the ones I had backed up on the cloud.

**Comment by MaxMusienko** [comment]
I have solved this problem! Deleted the cache file and was able to log into the program! Here is the link with which I found the solution! Good luck to everyone! 
https://monero.stackexchange.com/questions/3122/how-do-i-delete-the-wallet-cache/3123

**Comment by selsta** [comment]
@MaxMusienko make sure that you also have your seed backed up in case your whole hard drive fails

**Comment by sfsquires** [comment]
@MaxMusienko made an account just to say thanks for the link, really good timing u saved me a lot of headache

**Comment by monirgh1365-del** [comment]
Hello Thank you for your helpful guidance. I was getting an "invalid signature" error, which is why I followed your suggestion. However, now my wallet shows that my Monero balance is zero. How can I recover my Monero?

**Comment by monirgh1365-del** [comment]
> I have solved this problem! Deleted the cache file and was able to log into the program! Here is the link with which I found the solution! Good luck to everyone! https://monero.stackexchange.com/questions/3122/how-do-i-delete-the-wallet-cache/3123

Hello Thank you for your helpful guidance. I was getting an "invalid signature" error, which is why I followed your suggestion. However, now my wallet shows that my Monero balance is zero. How can I recover my Monero?

**Comment by nahuhh** [comment]
> Hello Thank you for your helpful guidance. I was getting an "invalid signature" error, which is why I followed your suggestion. However, now my wallet shows that my Monero balance is zero. How can I recover my Monero?

wait for the wallet to finish syncing
